dev/tasks/r/github.packages.yml

```diff
@@ -227,7 +227,7 @@ jobs:
            working-directory: 'arrow'
            extra-packages: cpp11
       - name: Set CRAN like openssl
-        if: contains(matrix.platform.runs_on, 'arm64')
+        if: contains(matrix.platform.name, 'arm64')

```

Comment:
I noticed both here and on main these jobs were [complaining about finding ssl](https://github.com/ursacomputing/crossbow/actions/runs/8866315145/job/24343853963#step:16:127) And it looks like this might be a typo here, I traced it back to #37684 where `runs_on` was split into `name` and `runs_on`. If I'm reading correctly what was here would never have run. 

I'm not sure how this was passing prior to the recent binary instability (and the windows failure mask when this ssl failure _started_). But this [seems to work](https://github.com/ursacomputing/crossbow/actions/runs/8868937137/job/24349295020) though the upload step [failed](https://github.com/ursacomputing/crossbow/actions/runs/8868937137/job/24349386706#step:11:72)

@assignUser (1) does this change look ok and (2) do you know what's up with the upload error?

Reply:
Maybe the failure is the change in version number? Or is it a permissions on from-fork builds (I don't _think so_ since it's actually from crossbow, but that could be passed through)?

Reply:
@jonkeane The change looks good but I think the issue actually happened when [we moved to gh hosted arm64](https://github.com/apache/arrow/pull/40437) runners (`['arm64', 'macos', 'self-hosted']` -> `'macos-14'`) , not sure why it didn't error then, probably some change in the image. Good catch!

The upload job is failing because `tasks.yml` still has an artifact named after the old pattern which causes the fail because it was never uploaded
![grafik](https://github.com/apache/arrow/assets/16141871/3669100b-6702-46c6-ba67-9a5ead7439c2)
I left a suggestion for that.